src/components/PeriodDimension/FixedPeriodSingleSelect.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from 'react'
+import propTypes from '@dhis2/prop-types'
+import i18n from '@dhis2/d2-i18n'
+import { SingleSelectField, SingleSelectOption } from '@dhis2/ui'
+import { spacers } from '@dhis2/ui-constants'
+import FixedPeriodFilter from './FixedPeriodFilter.js'
+import {
+    parsePeriodCode,
+    getFixedPeriodsOptionsById as getPeriodById,
+    getYearOffsetFromNow,
+} from './utils/fixedPeriods.js'
+
+const FixedPeriodSingleSelect = ({
+    onChange,
+    allowedPeriodTypes,
+    className,
+    dataTest,
+    value,
+}) => {
+    const [periodType, setPeriodType] = useState('')
+    const [year, setYear] = useState('')
+    const [options, setOptions] = useState()
+
+    useEffect(() => {
+        const period = parsePeriodCode(value, allowedPeriodTypes)
+        if (period) {
+            setPeriodType(period.id)
+            setYear(period.year)
+            setOptions(period.options)
+        }
+        if (!period && periodType && year) {
+            const offset = getYearOffsetFromNow(year)
+            const options = getPeriodById(periodType).getPeriods({ offset })
+            setOptions(options)
+        }
+    }, [allowedPeriodTypes, periodType, value, year])
+
+    return (
+        <div data-test={dataTest} className={className}>
+            <div className="row">
+                <FixedPeriodFilter
+                    allowedPeriodTypes={allowedPeriodTypes}
+                    currentPeriodType={periodType}
+                    currentYear={year}
+                    onSelectPeriodType={type => {
+                        setPeriodType(type)
+                        onChange('')
+                    }}
+                    onSelectYear={year => {
+                        setYear(year)
+                        onChange('')
+                    }}
+                />
+            </div>
+
+            {periodType && year && (
+                <SingleSelectField
+                    label={i18n.t('Period')}
+                    onChange={({ selected }) => onChange(selected)}
+                    dense
+                    selected={value}
+                >
+                    {options.map(option => (
+                        <SingleSelectOption
+                            key={option.id}
+                            value={option.id}
+                            label={option.name}
+                        />
+                    ))}
+                </SingleSelectField>
+            )}
+            <style jsx>{`
```

Comment:
Maybe nitpicking since it's such tiny bit, but it's preferred to keep the jsx in a separate `styles/[componentName].style.js` file.